### PR TITLE
feat: Resource to manage XSUAA Api Credentials

### LINKED ISSUE #689: [FEATURE] Managing API Credentials for Calling REST APIs of SAP Authorization and Trust Management Service
BTP CLI Server supports [Managing API Credentials for Calling REST APIs of SAP Authorization and Trust Management Service](https://help.sap.com/docs/btp/sap-business-technology-platform/managing-api-credentials-for-calling-rest-apis-of-sap-authorization-and-trust-management-service?locale=en-US&version=Cloud). Implement support for

- [ ] Create API credentials
- [ ] Show all API credentials
- [ ] Show details about specific API credentials
- [ ] Delete API credentials

### PR DESCRIPTION
**Pull request by diya-dhan**
## Purpose

fix #689 

The BTP CLI supports managing API credentials for the SAP Authorization and Trust Management (XSUAA) service, enabling users to interact programmatically with REST APIs. These APIs facilitate the management of entities and user provisioning across subaccounts, global accounts, or directories.

The provider now has introduced new resources to manage these API credentials at various levels: subaccount, global account, and directory. Users can create credentials of either the certificate type (currently, certificates can only be provided as strings) or the secret type. The state file will store the client secret or certificate key information for reference when using these APIs.

Please note that updating credentials is not supported by this resource, as there is currently no API available for this purpose.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code via automated test

```bash
go test ./...
```

## What to Check

Verify that the following are valid:

* Automated tests are executed successfully

## Checklist for reviewer

The following organizational tasks must be completed before merging this PR:

* [x] The PR is assigned to the Terraform project and a status is set (typically "in review").
* [x] The PR has the matching labels assigned to it.
* [x] The PR has a milestone assigned to it.
* [x] If the PR closes an issue, the issue is referenced.
* [x] Possible follow-up items are created and linked.

### CODE REVIEW

File: examples/resources/btp_directory_api_credential/import.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# terraform import btp_directory_api_credential.<resource_name> <directory_id>
```
Comment (lechnerc77) [inline examples/resources/btp_directory_api_credential/import.sh]: The comment in the import should contain the full key for import. It seems that here this is only a part.
If it is a mixed key, then the different components should be comma.-separated without a space. See e.g. <https://registry.terraform.io/providers/SAP/btp/latest/docs/resources/directory_role_collection#import>

File: examples/resources/btp_globalaccount_api_credential/import.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# terraform import btp_globalaccount_api_credential.<resource_name> <globalaccount_id>
```
Comment (lechnerc77) [inline examples/resources/btp_globalaccount_api_credential/import.sh]: The comment in the import should contain the full key for import. It seems that here this is only a part.
If it is a mixed key, then the different components should be comma.-separated without a space. See e.g. <https://registry.terraform.io/providers/SAP/btp/latest/docs/resources/directory_role_collection#import>

File: examples/resources/btp_subaccount_api_credential/import.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# terraform import btp_subaccount_api_credential.<resource_name> <subaccount_id>
```
Comment (lechnerc77) [inline examples/resources/btp_subaccount_api_credential/import.sh]: The comment in the import should contain the full key for import. It seems that here this is only a part.
If it is a mixed key, then the different components should be comma.-separated without a space. See e.g. <https://registry.terraform.io/providers/SAP/btp/latest/docs/resources/directory_role_collection#import>

**Review by lechnerc77 — CHANGES_REQUESTED** [review]
Some minor changes requested for import documentation

General question: there are no data sources available. This is due to the fact that the returned data is only a fraction of the complete data and we leave it out?

**Comment by lechnerc77** [comment]
Sonar Cloud Quality check is accepted to be below 80%

**Comment by diya-dhan** [comment]
> Some minor changes requested for import documentation
> 
> General question: there are no data sources available. This is due to the fact that the returned data is only a fraction of the complete data and we leave it out?

Yes, like you mentioned the GET call does not return a lot of the information that is relevant to the user. This is only known at the time the api credentials are created. Additionally, as of now, there is no use case where the api credentials can be used in other BTP resources, due to which a datasource for it didn't really make sense.

**Review by lechnerc77 — APPROVED** [review]
lgtm

**Pull request merged**